liver_imaging_analysis/engine/engine.py

```diff
@@ -428,62 +536,166 @@ def test(self, dataloader = None, callback = False):
         float
             the averaged metric calculated during testing
         """
-        if dataloader is None: #test on test set by default
+        if dataloader is None:  # test on test set by default
             dataloader = self.test_dataloader
         num_batches = len(dataloader)
         test_loss = 0
         test_metric = 0
         self.network.eval()
         with torch.no_grad():
-            for batch_num,batch in enumerate(dataloader):
+            for batch_num, batch in enumerate(dataloader):
                 batch[Keys.IMAGE] = batch[Keys.IMAGE].to(self.device)
                 batch[Keys.LABEL] = batch[Keys.LABEL].to(self.device)
                 batch[Keys.PRED] = self.network(batch[Keys.IMAGE])
-                test_loss += self.loss(
-                    batch[Keys.PRED],
-                    batch[Keys.LABEL]
-                    ).item()
-                #Apply post processing transforms on prediction
+                test_loss += self.loss(batch[Keys.PRED], batch[Keys.LABEL]).item()
+                # Apply post processing transforms on prediction
                 batch = self.post_process(batch)
                 self.metrics(batch[Keys.PRED].int(), batch[Keys.LABEL].int())
                 if callback:
-                  self.per_batch_callback(
-                      batch_num,
-                      batch[Keys.IMAGE],
-                      batch[Keys.LABEL],
-                      batch[Keys.PRED]
-                      )
+                    self.per_batch_callback(
+                        batch_num,
+                        batch[Keys.IMAGE],
+                        batch[Keys.LABEL],
+                        batch[Keys.PRED],
+                    )
             test_loss /= num_batches
             # aggregate the final metric result
             test_metric = self.metrics.aggregate().item()
             # reset the status for next computation round
             self.metrics.reset()
         return test_loss, test_metric
 
+    def objective(
+        self,
```

Comment:
add default values for non mandatory args

Reply:
done